flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/utils/LegacyTypeInfoDataTypeConverter.java

```diff
@@ -181,6 +182,10 @@ else if (typeInfo instanceof CompositeType) {
 			return createLegacyType(LogicalTypeRoot.STRUCTURED_TYPE, typeInfo);
 		}
 
+		else if (typeInfo instanceof GenericTypeInfo) {
```

Comment:
Can you add test to `LegacyTypeInfoDataTypeConverterTest`?

Reply:
`LegacyTypeInfoDataTypeConverterTest` already covers RAW type, I updated the test result.